dotty/matchers-core/src/main/scala/org/scalatest/matchers/CompileMacro.scala

```diff
@@ -25,12 +25,12 @@ import scala.quoted._
 object CompileMacro {
 
   // check that a code snippet compiles
-  def assertCompileImpl[T](self: Expr[T], compileWord: Expr[CompileWord], pos: Expr[source.Position])(shouldOrMust: String)(using Quotes): Expr[Assertion] = {
+  def assertCompileImpl[T](self: Expr[T], typeChecked: Expr[Boolean], compileWord: Expr[CompileWord], pos: Expr[source.Position])(shouldOrMust: String)(using Quotes): Expr[Assertion] = {
     import quotes.reflect._
 
     // parse and type check a code snippet, generate code to throw TestFailedException if both parse and type check succeeded
     def checkCompile(code: String): Expr[Assertion] =
-      if (/*typeChecks(code)*/ true) '{ Succeeded } // FIXME
+      if (typeChecked.valueOrError) '{ Succeeded } // FIXME
```

Comment:
@cheeseng If this was fixed, then next round please delete the FIXME comment. Tnx.

Reply:
@bvenners Sure, I'll do in the next-batch branch that I am working on currently.

Thanks.